tests/performance/timing_navier_stokes.cc

```diff
@@ -0,0 +1,2298 @@
+
+        default:
+          Assert(false, ExcNotImplemented());
+      }
+
+    switch (get_testing_environment())
+      {
+        case TestingEnvironment::light:
+          triangulation.refine_global(1);
+          break;
+        case TestingEnvironment::medium:
+          triangulation.refine_global(2);
+          break;
+        case TestingEnvironment::heavy:
+          triangulation.refine_global(3);
+          break;
+      }
+  }
+
+
+
+  template <int dim>
+  void
+  FlowProblem<dim>::output_results(const unsigned int)
+  {
+    Postprocessor postprocessor;
+    DataOut<dim>  data_out;
+
+    DataOutBase::VtkFlags flags;
+    flags.write_higher_order_cells = true;
+    data_out.set_flags(flags);
+
+    data_out.attach_dof_handler(dof_handler);
+    {
+      std::vector<std::string> names;
+      names.emplace_back("density");
+      for (unsigned int d = 0; d < dim; ++d)
+        names.emplace_back("momentum");
+      names.emplace_back("energy");
+
+      std::vector<DataComponentInterpretation::DataComponentInterpretation>
+        interpretation;
+      interpretation.push_back(
+        DataComponentInterpretation::component_is_scalar);
+      for (unsigned int d = 0; d < dim; ++d)
+        interpretation.push_back(
+          DataComponentInterpretation::component_is_part_of_vector);
+      interpretation.push_back(
+        DataComponentInterpretation::component_is_scalar);
+
+      data_out.add_data_vector(dof_handler, solution, names, interpretation);
+    }
+    data_out.add_data_vector(solution, postprocessor);
+
+    LinearAlgebra::distributed::Vector<Number> reference;
+    if (testcase == 0 && dim == 2)
+      {
+        reference.reinit(solution);
+        flow_operator.project(ExactSolution<dim>(time), reference);
+        reference.sadd(-1., 1, solution);
+        std::vector<std::string> names;
+        names.emplace_back("error_density");
+        for (unsigned int d = 0; d < dim; ++d)
+          names.emplace_back("error_momentum");
+        names.emplace_back("error_energy");
+
+        std::vector<DataComponentInterpretation::DataComponentInterpretation>
+          interpretation;
+        interpretation.push_back(
+          DataComponentInterpretation::component_is_scalar);
+        for (unsigned int d = 0; d < dim; ++d)
+          interpretation.push_back(
+            DataComponentInterpretation::component_is_part_of_vector);
+        interpretation.push_back(
+          DataComponentInterpretation::component_is_scalar);
+
+        data_out.add_data_vector(dof_handler, reference, names, interpretation);
+      }
+
+    Vector<double> mpi_owner(triangulation.n_active_cells());
+    mpi_owner = Utilities::MPI::this_mpi_process(MPI_COMM_WORLD);
+    data_out.add_data_vector(mpi_owner, "owner");
+
+    data_out.build_patches(mapping,
+                           fe.degree,
+                           DataOut<dim>::curved_inner_cells);
+
+    // Do not write a file here to be independent of file system
```

Comment:
Does it make sense to keep graphical output, schlieren, etc. in this benchmark?

Reply:
I am split about this one. I think it would be good to have the `build_patches` code in, which steps into a typical postprocessor and might honour attempts to improve those functions (or alert us when we make them worse).

One issue I should probably fix is the fact that the benchmark contains some dead code, which it inherits from the step-76 program. If you prefer, I can remove all dead code and only keep the part relevant for this benchmark (the three-dimensional Taylor-Green vortex, which is originally an incompressible flow but is here run with a low Mach number of 0.1).

Do you agree on those two points?

Reply:
Let me add that I do not think we should have the `write_vtu_in_parallel` part because it depends on the file system and because it produces a non-negligible amount of data (I think several hundreds of MB for the `large` configuration).

Reply:
I agree with not writing output and it makes sense to keep build_patches. If you are reasonably sure that the current chosen test works well, I would also remove any dead code unless it is something that might help with debugging.